### PR TITLE
[DUOS-1008] Standardize UserRoleDAO

### PR DESCRIPTION
**Pull request by hams7504**
### Addresses
https://broadworkbench.atlassian.net/browse/DUOS-1008

### Summary
- Checked that all methods in UserRoleDAO were tested
- Updated all SQL queries
- Changed column name in roles table

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/db/UserRoleDAO.java [hunk 1]
```diff
@@ -18,46 +18,81 @@
...
       @Bind("userId") Integer userId,
       @Bind("existentRoleId") Integer existentRoleId);
 
-  @SqlUpdate("delete from user_role where user_id = :userId and role_id IN (<existentRoles>)")
+  @SqlUpdate("""
+    DELETE FROM user_role WHERE user_id = :userId AND role_id IN (<existentRoles>)
+    """)
   void removeUserRoles(@Bind("userId") Integer userId,
-      @BindList("existentRoles") List<Integer> existentRoles);
+      @BindList("existentRoles") List<Integer> existentRoles); // what is existent_roles?
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/db/UserRoleDAO.java]: `existentRoles` is just the variable name for the role ids that the user may have.

File: src/main/java/org/broadinstitute/consent/http/db/UserRoleDAO.java [hunk 2]
```diff
@@ -18,46 +18,81 @@
...
-  @SqlQuery("select * from user_role ur inner join roles r on r.roleId = ur.role_id where ur.user_id = :userId and ur.role_id = :roleId")
+  @SqlQuery("""
+    SELECT *
+    FROM user_role ur 
+    INNER JOIN roles r ON r.role_id = ur.role_id
+    WHERE ur.user_id = :userId AND ur.role_id = :roleId
+    """)
   UserRole findRoleByUserIdAndRoleId(@Bind("userId") Integer userId,
-      @Bind("roleId") Integer roleId);
+      @Bind("roleId") Integer roleId); // change r.roleId
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/db/UserRoleDAO.java]: nit: rm comment.

File: src/main/resources/changelog-master.xml [hunk 3]
```diff
@@ -128,4 +128,6 @@
     relativeToChangelogFile="true"/>
   <include file="changesets/changelog-consent-2023-07-27-match-rationale.xml"
     relativeToChangelogFile="true"/>
+  <include file="changesets/changelog-consent-2023-09-20-capitalize-datasets-used-in-user-role-dao.xml"
+    relativeToChangelogFile="true"/>
```
Comment (rushtong) [inline src/main/resources/changelog-master.xml]: nit: I think this name is misleading, all we're really doing here is renaming `roleid` -> `role_id`

**Review by rushtong — COMMENTED** [review]
This looks good! See minor comments inline.

File: src/main/resources/changelog-master.xml (hunk 3 above)
Comment (hams7504) [reply src/main/resources/changelog-master.xml]: Done!

**Review by rushtong — APPROVED** [review]
Thank you for the changes 👍🏽

**Review by fboulnois — APPROVED** [review]
looks great 👍

**Pull request merged**